Documentation/design-docs/DotNetCore-SharedPackageStore.md

```diff
@@ -0,0 +1,219 @@
+
+Note that the `profile.xml` specifies exact RID-specific or IL packages to filter out of `dotnet publish` output -- i.e., the physical files will be filtered out of the `dotnet publish` output. The `deps.json` (logical assets) file will still contain the entries specified in the `profile.xml` file.
+
+## Scenarios
+
+### ASP .NET
++ If authoring shared package installers (eager cache):
+    - Start from a clean directory and a list of packages in `packages.xml` file.
+    - Use `dotnet cache` to produce the layout in the directory.
+    - Make MSI/pkg/deb and zips of this layout.
+    - Publish the `profile.xml` file that users can use in sync with the installers.
+    - Developer/Deployment-admin installs the MSIs/zips to the deployment machines.
++ If letting app deployers cache shared packages (lazy cache):
+    - Publish `packages.xml` file that can be used to perform `dotnet cache`.
+    - Publish `profile.xml` file that can perform publish filtering.
+    - Deployment-admin issues `dotnet cache` when running the app.
++ Developer/Deployment-admin issues `dotnet publish filter profile.xml` to produce an ASP.NET app without containing shared components.
+   - Or developer/Deployment-admin issues `dotnet run` after installing MSIs or zips.
+
+### Antares
++ Antares produces the layout using a list of packages and `dotnet cache` in a folder.
++ This folder is then chained into environment variable: `DOTNET_SHARED_PACKAGES`.
++ When building app from source, issue `dotnet run` to pick up the shared packages.
++ When publishing an app to run, issue `dotnet publish filter profile.xml` with Antares profile.
+
+### Roslyn
++ Roslyn packages are part of the packages directory
++ Apps that use Roslyn must explicitly take a dependency on Roslyn and not through M.N.A.
++ Apps can choose to skip publishing these DLLs using the Roslyn profile file that we'd publish.
++ Apps are deployed on another machine where the runtime is installed.
++ Apps will run as Roslyn packages are part of the runtime installer.
+
+### Hosting Primers
++ I host apps that are already published
+    - Use `dotnet cache` and produce layout or unzip an earlier layout.
+    - Set `DOTNET_SHARED_PACKAGES` to point to layout.
+    - Nature of publish:
+        + Using my hosting profile
+          + App publish directory doesn't contain the filtered files picked from layout.
+        + Without my hosting profile
+          + Assemblies from app publish directory are overridden (*status quo*)
+
++ I build user apps from source
+    - dotnet cache
+    - Zip the layout
+    - Deploy on hosting servers
+    - Publish apps with filter
+
+## Work Items
+
+### Core-Setup
+
++ Change name of `Microsoft.NETCore.App` to `netcoreapp2.0` in the dotnet root. Note that this has to maintain compat.
++ Move the Roslyn assemblies out of the shared framework.
++ Build installers on Windows, Ubuntu and OSX for the Roslyn (shared) packages.
```

Comment:
We aren't doing this anymore.  Roslyn will be included in the ASP.NET runtime installer.